Source/Magick.NET.CrossPlatform/ubuntu.16.04/Dockerfile

```diff
@@ -2,20 +2,49 @@ FROM ubuntu:16.04
 RUN sed -i -e 's/#PasswordAuthentication yes/PasswordAuthentication no/g' /etc/ssh/sshd_config
-RUN mkdir ~/.ssh;echo "ssh-rsa AAAAB3NzaC1yc2EAAAABJQAAAQEAwFyerjwW+MpAIV056qHP0pJLAJn2acRQNtFnmQiCXP6AukRMr7rJ23w3TsyGEKzTGCi/jQMHTWzGYGj7OtNNTN+OSFvx9OVg+4wr6kYPQmPZBnPvUg727nqggAGZ90j7bL/B+7qiUJxoHOBqPSS56A7oMmsA19Iy2kGaYlBLB0J+YY1wHlRyVa4vyxZox6n9ORZ6Wat+OGc6fyo9sNYWBCVzufUWAhQM2JCctCO5htglytD3vk/wJnJ5k/VEajYueM90m3V7fv4DucWvPJk75TlSJUkh81yujQttPmar74k4vcTf/zfUpkcYBHucvMATUrBfpE+e1OL013JM470RuQ==" > ~/.ssh/authorized_keys;chmod 600 ~/.ssh/authorized_keys
+COPY authorized_keys /root/.ssh/authorized_keys
+RUN chmod 600 ~/.ssh/authorized_keys
+
+# Build libbz2
+RUN cd ~; wget -q http://www.bzip.org/1.0.6/bzip2-1.0.6.tar.gz; tar xf bzip2-1.0.6.tar.gz;
+RUN cd ~/bzip2-1.0.6; make CFLAGS="-Wall -Winline -O3 -fPIC -g -D_FILE_OFFSET_BITS=64"; make install;
+
+# Build zlib
+RUN git clone -n https://github.com/madler/zlib.git ~/libz; cd ~/libz; git checkout tags/v1.2.11
+RUN cd ~/libz; export CFLAGS="-O3 -fPIC"; ./configure --static; make; make install
+
+# Build lzma
+RUN git clone -n https://git.tukaani.org/xz.git ~/lzma; cd ~/lzma; git checkout tags/v5.2.3
+RUN cd ~/lzma; autoreconf -fiv; export CFLAGS="-O3 -fPIC"; ./configure; make; make install;
+
+# Build libjpeg-turbo
+RUN git clone -n https://github.com/libjpeg-turbo/libjpeg-turbo.git ~/libjpeg; cd ~/libjpeg; git checkout tags/1.5.3
+RUN cd ~/libjpeg; autoreconf -fiv; ./configure --with-jpeg8 CFLAGS="-O3 -fPIC"; make; make install prefix=/usr/local libdir=/usr/local/lib64
+
+# Build libpng16
+RUN git clone -n https://github.com/glennrp/libpng.git ~/libpng; cd ~/libpng; git checkout tags/v1.6.34
+RUN cd ~/libpng; autoreconf -fiv; ./configure CFLAGS="-O3 -fPIC"; make; make install
+
+# Build libtiff
+RUN git clone -n https://gitlab.com/libtiff/libtiff.git ~/libtiff; cd ~/libtiff; git checkout tags/Release-v4-0-9
+RUN cd ~/libtiff; autoreconf -fiv; export CFLAGS="-O3 -fPIC"; ./configure; make; make install
+
+# Build libwebpmux/demux
+RUN git clone -n https://chromium.googlesource.com/webm/libwebp ~/libwebp; cd ~/libwebp; git checkout tags/v0.6.1
+RUN cd ~/libwebp; autoreconf -fiv; export CFLAGS="-O3 -fPIC"; ./configure --enable-libwebpmux --enable-libwebpdemux; make; make install;
+
 
 # Build ImageMagick
-RUN mkdir Magick.NET
-RUN git clone --depth 1 https://github.com/dlemstra/Magick.NET.git Magick.NET
-RUN cd Magick.NET/ImageMagick/Source;bash Checkout.sh
-RUN cd Magick.NET/ImageMagick/Source/ImageMagick/ImageMagick;./configure --with-magick-plus-plus=no --with-quantum-depth=8 --enable-hdri=no;make install
-RUN cd Magick.NET/ImageMagick/Source/ImageMagick/ImageMagick;./configure --with-magick-plus-plus=no --with-quantum-depth=16 --enable-hdri=no;make install
-RUN cd Magick.NET/ImageMagick/Source/ImageMagick/ImageMagick;./configure --with-magick-plus-plus=no --with-quantum-depth=16;make install
+RUN git clone -n https://github.com/ImageMagick/ImageMagick ~/ImageMagick
```

Comment:
I wonder if we could use this script `ImageMagick/Source/Checkout.sh` here instead. Maybe we could use that for all the repos? I do think I need to change that script and use a git sha instead and checkout the other repos (dependencies) based on the date of that checkout. Would be very nice if you could change that script for me and use it here.

Reply:
How would you feel about just using git submodules? We use them extensively in some internal projects with similar structure and it works pretty well for us.

Reply:
Not a fan of submodules would prefer to use a script.

Reply:
Fair enough. I am going to play with a couple other alternative approaches, and worst case I will just plug the existing checkout script into the dockerfile and see how it goes.

Reply:
I think doing the actual checkouts during the docker build process is just too problematic. It's way too easy to mix up versions, and way to hard to actually work in branches or forks without tripping all over yourself (not to mention the havoc it wreaks with docker layer caching).

To get around that, I think the best approach will be to just move the dockerfile from the `Source\Magick.NET.CrossPlatform\ubuntu.16.04` folder over into the `ImageMagick\Source` folder. Checkouts then can happen in your working copy like normal, via the Checkout.sh script or manually, and the docker build context can just include them from there without any extra fuss. I have this mostly working, and so far only the ImageMagick/png project has needed any changes applied to get the whole build chain working again.

Do you have any opposition to including liblzma? If we want to stick to the pattern it would need a fork set up under the ImageMagick account, from https://git.tukaani.org/xz.git.